### PR TITLE
Add plugin overrides configuration to flyte-cli

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
Add plugin overrides configuration to flyte-cli.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
https://docs.google.com/document/d/1OWYJ2ggr0pWvOdOKkf04NLtj2Rf32dMVYvAYMoyeU48/edit#

## Tracking Issue
https://github.com/lyft/flyte/issues/499

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by katrogan** [comment]
N.B. I won't submit this until the corresponding admin change is merged

File: tests/flytekit/unit/models/test_matchable_resource.py
Comment (kumare3) [inline tests/flytekit/unit/models/test_matchable_resource.py:48]: so we can dynamically alter this? Is this desirable? I guess it is good to have here, but we should definitely record the originator of this request

Comment (katrogan) [reply tests/flytekit/unit/models/test_matchable_resource.py:48]: yeah every request in admin is audited (now that we semi-have auth) so this should in theory be traceable

**Comment by katrogan** [comment]
friendly ping @wild-endeavor

**Review by wild-endeavor — DISMISSED** [review]

**Comment by katrogan** [comment]
ping @wild-endeavor

**Review by wild-endeavor — APPROVED** [review]

**Pull request merged**
